### ISSUE TITLE
Flaky test TestMemoryConnectorTest.testCustomMetricsScanFilter

### ISSUE THREAD

**Issue by findepi** [trinodb/trino#8691]
https://github.com/trinodb/trino/runs/3179633364

```
Error:  testCustomMetricsScanFilter(io.trino.plugin.memory.TestMemoryConnectorTest)  Time elapsed: 0.125 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<LongCount{total=[2000]}> but was:<LongCount{total=[1001]}>
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at io.trino.plugin.memory.TestMemoryConnectorTest.testCustomMetricsScanFilter(TestMemoryConnectorTest.java:156)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

cc @rzeyde-varada

**Comment by findepi** [comment]
https://github.com/trinodb/trino/runs/3181603026 cc @rzeyde-varada

**Comment by rzeyde-varada** [comment]
Reproduced locally (failed 2 out of 10k runs of `testCustomMetricsScanFilter`)
```
org.junit.ComparisonFailure: 
Expected :LongCount{total=2000}
Actual   :LongCount{total=1502}

	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at io.trino.plugin.memory.TestMemoryConnectorTest.testCustomMetricsScanFilter(TestMemoryConnectorTest.java:154)
```

**Comment by rzeyde-varada** [comment]
`testCustomMetricsScanOnly` also failed locally 3 times (out of 10k runs) with:
```
 org.junit.ComparisonFailure: 
Expected :LongCount{total=2000}
Actual   :LongCount{total=1162}
(happened twice)

org.junit.ComparisonFailure: 
Expected :LongCount{total=2000}
Actual   :LongCount{total=1834}
(happened once)
```

**Comment by rzeyde-varada** [comment]
It seems that also `OperatorStats#getInputPositions` (which is a "non-custom metric") aggregation returns the wrong result when the above error occurs:
```
inputRows: 1834
metrics: {rows=LongCount{total=1834}, finished=LongCount{total=8}, started=LongCount{total=8}}

org.junit.ComparisonFailure: 
Expected :LongCount{total=2000}
Actual   :LongCount{total=1834}
```
I have used the following snippet to compute the total input positions:
https://github.com/rzeyde-varada/trino/commit/4399126e49aa6b0939f3ae3c4fbdbd391b908efd#diff-920e9ca800575089f913cc4063da9962bfe7bd4adbd617668c973a7004cd2c99R178

**Comment by findepi** [comment]
Is it another incarnation of https://github.com/trinodb/trino/issues/5172?
We have marked many tests as `@Flaky` attributing flakiness to that issue.

**Comment by findepi** [comment]
cc @sopel39

**Comment by findepi** [comment]
see https://github.com/trinodb/trino/pull/8739#issuecomment-890960099

**Comment by findepi** [comment]
https://github.com/trinodb/trino/runs/3409206985

**Comment by findepi** [comment]
https://github.com/trinodb/trino/runs/3410107887

**Comment by findepi** [comment]
https://github.com/trinodb/trino/runs/3614059303

**Comment by findepi** [comment]
https://github.com/trinodb/trino/runs/3636321727

**Comment by findepi** [comment]
https://github.com/trinodb/trino/runs/3666711537 cc @sopel39 @raunaqmorarka

**Comment by ebyhr** [comment]
https://github.com/trinodb/trino/runs/3835521895

**Comment by wendigo** [comment]
https://github.com/trinodb/trino/pull/9517/checks

**Comment by findepi** [comment]
>  closed this in #9336 18 days ago

seems it wasn't sufficient, per above

**Comment by ebyhr** [comment]
https://github.com/trinodb/trino/runs/3876699524
